cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/status/StatusModule.java

```diff
@@ -51,11 +53,16 @@ public StatusModule(ModuleConfiguration configuration, List<Reporter> reporters)
         StatusConfiguration config = StatusConfiguration.create(configuration.options);
         period = config.period();
         timeunit = config.timeunit();
+        compactionsEnabled = config.compactionsEnabled();
 
         infoProvider = DiagnosticsAgent.getInfoProvider();
-
-        timer = new Timer(STATUS_THREAD_NAME);
-        timer.scheduleAtFixedRate(new StatusTask(), 0, config.reportingRateInMillis());
+        if (infoProvider == null) {
+            logger.warn("Failed to initialize StatusModule. Info provider is null");
+            timer = null;
+        } else {
```

Comment:
Why would you schedule timer if `compactionEnabled = false`, should we move `if(compactionEnabled)` here? This way every minute it will check if it is enabled in some thread.